File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/Ocg.java
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel;

import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Builder
@Data
@EqualsAndHashCode(of = { "ocgId"})
@ToString
public class Ocg implements RiskDataSet {
    public static int OCG_ID_POSITION = 0;
    public static int OCGM_BAND_POSITION = 1;

    private String ocgId;
    private String ocgmBand;

    public String getKey() {
        return ocgId;
    }

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/Ocg.kt
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel

data class Ocg(private val ocgId: String) : RiskDataSet {

  constructor(ocgId: String, ocgmBand: String?) : this(ocgId) {
    this.ocgmBand = ocgmBand
  }

  var ocgmBand: String? = null

  override fun getKey(): String {
    return ocgId
  }

  companion object {
    @JvmField
    var OCG_ID_POSITION = 0

    @JvmField
    var OCGM_BAND_POSITION = 1
  }
}
